### PR TITLE
Removes locking from step()

### PR DESCRIPTION
**Pull request by gilwoolee**
`KinematicSimulationTrajectoryExecutor::step()` should no longer lock the skeleton and let the caller be responsible for locking if it's running under multi-threaded setting. This is consistent with other executors.

### CODE REVIEW

File: include/aikido/control/BarrettHandKinematicSimulationPositionCommandExecutor.hpp [hunk 1]
```diff
@@ -47,7 +47,9 @@ class BarrettHandKinematicSimulationPositionCommandExecutor
   /// \return Future which becomes available when the execution completes.
   std::future<void> execute(const Eigen::VectorXd& goalPositions) override;
 
-  // Documentation inherited.
+  /// If multiple threads are accessing this function or the skeleton associated
+  /// with this executor, it is necessary to lock the skeleton before
+  /// calling this method.
```
Comment (jslee02) [inline include/aikido/control/BarrettHandKinematicSimulationPositionCommandExecutor.hpp]: Nit: I believe this comment is a note to be _added_ to the original docstring of `PositionCommandExecutor::step()`, but this will just overwrite the original docstring. It would be nice to copy the original docstring and add this node as:
```cpp
  /// \copydoc PositionCommandExecutor::step()
  ///
  /// If multiple threads are accessing this function or the skeleton associated
  /// with this executor, it is necessary to lock the skeleton before
  /// calling this method.
```

Also, it would be great to actually add a docstring to `PositionCommandExecutor::step()`. Currently, it's `// Step once`. If it's enough then make it doxygen comment by prepending one more `/`.

File: include/aikido/control/KinematicSimulationTrajectoryExecutor.hpp [hunk 2]
```diff
@@ -35,7 +35,9 @@ class KinematicSimulationTrajectoryExecutor : public TrajectoryExecutor
   std::future<void> execute(
     trajectory::TrajectoryPtr traj) override;
 
-  // Documentation inherited.
+  /// If multiple threads are accessing this function or the skeleton associated
+  /// with this executor, it is necessary to lock the skeleton before
+  /// calling this method.
```
Comment (jslee02) [inline include/aikido/control/KinematicSimulationTrajectoryExecutor.hpp]: ditto

File: src/control/KinematicSimulationTrajectoryExecutor.cpp [hunk 3]
```diff
@@ -106,7 +106,7 @@ void KinematicSimulationTrajectoryExecutor::step()
   }
 
   auto timeSinceBeginning = system_clock::now() - mExecutionStartTime;
-  double t = duration_cast<seconds>(timeSinceBeginning).count();
+  auto t = duration<double>(timeSinceBeginning).count();
```
Comment (jslee02) [inline src/control/KinematicSimulationTrajectoryExecutor.cpp]: Nit: I believe the unit of time here is seconds. It would be nice to either (1) change the variable name to encode the unit or (2) add trailing comment something like `// sec`.

Additionally, it would be much better to comment the unit of the time parameter to the docstring of `Trajectory::evaluate()`.

File: src/control/KinematicSimulationTrajectoryExecutor.cpp [hunk 4]
```diff
@@ -129,6 +126,7 @@ void KinematicSimulationTrajectoryExecutor::step()
     mPromise->set_value();
     mInExecution = false;
   }
+
```
Comment (jslee02) [inline src/control/KinematicSimulationTrajectoryExecutor.cpp]: Nit: Could we remove this empty line?

File: src/control/KinematicSimulationTrajectoryExecutor.cpp (hunk 3 above)
Comment (mkoval) [inline src/control/KinematicSimulationTrajectoryExecutor.cpp]: Why was this changed to `duration` instead of `duration_cast?` I believe `timeSinceBeginning` is a duration, so `duration_cast` was appropriate here. 🤔

File: src/control/KinematicSimulationTrajectoryExecutor.cpp [hunk 5]
```diff
@@ -105,8 +105,9 @@ void KinematicSimulationTrajectoryExecutor::step()
     mInExecution = false;
   }
 
+  using seconds = std::chrono::duration<float>;
```
Comment (jslee02) [inline src/control/KinematicSimulationTrajectoryExecutor.cpp]: Nit: I believe the type of time should be `double` as `Trajectory` takes it as `double`.

File: include/aikido/trajectory/Trajectory.hpp [hunk 6]
```diff
@@ -50,8 +50,8 @@ class Trajectory
   /// function is implementation-defined if \c _t is not between
   /// \c getStartTime() and \c getEndTime().
   ///
-  /// \param _t time parameter
-  /// \param[out] _state output state of the trajectory at time \c _t
+  /// \param _t time parameter (in seconds)
+  /// \param[out] _state output state of the trajectory at time \c _t (seconds)
   virtual void evaluate(double _t,
     statespace::StateSpace::State *_state) const = 0;
```
Comment (mkoval) [inline include/aikido/trajectory/Trajectory.hpp]: This isn't really "in seconds" -it's an arbitrary path parameter.

File: src/control/KinematicSimulationTrajectoryExecutor.cpp [hunk 7]
```diff
@@ -105,8 +105,9 @@ void KinematicSimulationTrajectoryExecutor::step()
     mInExecution = false;
   }
 
+  using seconds = std::chrono::duration<double>;
```
Comment (mkoval) [inline src/control/KinematicSimulationTrajectoryExecutor.cpp]: This is a potentially confusing alias, because `std::chrono::seconds` stores integers and this alias stores `double`s.
